### DISCUSSION TITLE
Feature Request: Add an Exclusion List for Tab Sleeping

### DISCUSSION THREAD

**Discussion by Seele0oO** [zen-browser/desktop#3118] [Ideas]
Currently, ZenBrowser's tab sleeping feature helps reduce resource usage, but in some cases, certain websites need to remain active to ensure compatibility and usability (e.g., real-time updating pages or frequently used tool-based websites). It would be helpful to add an exclusion list for the tab sleeping feature, allowing users to manually specify domains that are exempt from being put to sleep.

Current Behavior
I couldn't find an option for this functionality in the current settings (if it already exists, please let me know).

Expected Behavior

    Users can manually add specific domains/URLs to an exclusion list through the settings or extension configuration.
    Domains/URLs in the exclusion list will never be suspended by the tab sleeping feature.
    Optionally, support for regex patterns for more advanced matching (if feasible).

Additional Notes
This feature would enhance compatibility with real-time tools and applications while making the tab sleeping feature more flexible and user-friendly.

This is an alternative solution to [this problem](https://github.com/zen-browser/desktop/issues/2479), but I think it deserves to be rolled out as a separate feature.

**Comment by tranquil-tr0** [comment]
Hi! I think this may be solved through the `zen.tab-unloader.excluded-urls` option in `about:config`, or in the tab context menu (that opens upon right-click) if you would like to only target a single tab (I'm not too sure about this option, since I turned off the Zen tab unloader). Let me know if this works/doesn't work!
Side note: although, this could probably be in the settings menu.

**Reply by Fxxxlei** [reply]
I think this warrants a dedicated setting in the settings menu. It's a setting you might want to change regularly and many people might have a good reason to use.

**Reply by tranquil-tr0** [reply]
Not sure about other browsers, but I know Brave has that setting so it does seem to exist otherwise (as in, in other browsers).

**Comment by kezyr** [comment]
Hi! Just for importance increasing of this topic/setting ability. For example I cannot use multiple instances of website https://godbolt.org/ because if the tab goes to sleep then after wake up the content that was in editor at that specific instance is deleted and instead of the original content there is a copy from different instance. I dont know whether the website or browser is responsible for this but it is a big blocker for me. I am going to test with `zen.tab-unloader.excluded-urls`.

**Reply by kezyr** [reply]
I confirm using `zen.tab-unloader.excluded-urls` works well. If I dont care about my memory, how can I completely turn off tab sleeping?

**Reply by vraravam** [reply]
do you know if it can handle regex? like `*.google.com`?

**Reply by kezyr** [reply]
It looks that regex is not supported. I tried *.org and wikipedia.org and support.mozilla.org were unloaded

**Reply by kezyr** [reply]
So. Regex does not work but for example placing just `.org` works. there is also `zen.tab-unloader.enabled` that can be changed to `false` and then the tab sleeping (correct name is tab unloading) is disabled. But what I like I found is `zen.tab-unloader.timeout-minutes` that is at default set to 20 minutes and it can be changed. So everything can be fine tuned at least there
